src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrScanner.cs

```diff
@@ -79,12 +82,59 @@ private async Task<string[]> ScanRepository(SourceMapping sourceMapping, Cancell
 
         foreach (var file in files)
         {
-            _logger.LogWarning("File {file} is cloaked but present in the VMR", file);
+            _logger.LogWarning("File in {file} is cloaked but present in the VMR", file);
         }
 
         return files;
     }
 
-    private static string ExcludeFile(string file) => $":(attr:!{VmrInfo.KeepAttribute}){file}";
+    private static string ExcludeFile(string file, string gitAttribute) => $":(attr:!{gitAttribute}){file}";
+
+    public async Task<List<string>> ListBinaryFiles(CancellationToken cancellationToken)
```

Comment:
First thing I am thinking about - why not make 2 scanner classes? Maybe with a common ancestor. I am fairly sure these will keep growing.

Reply:
Done, they do have a common ancestor, as they share a lot of the same code. I also did a similar thing for the operation classes